core/virtualhostname/virtualhostname_test.go

```diff
@@ -100,8 +93,159 @@ func (s *HostnameSuite) TestParseHostname(c *gc.C) {
 		if tC.expectedErr == "" {
 			c.Assert(err, gc.IsNil)
 			c.Assert(res, gc.DeepEquals, tC.result)
+			c.Assert(res.String(), gc.Equals, tC.hostname)
+		} else {
+			c.Assert(err, gc.ErrorMatches, tC.expectedErr)
+		}
+	}
+}
+
+func (s *HostnameSuite) TestNewInfoMachineTarget(c *gc.C) {
+	testCases := []struct {
+		desc        string
+		modelUUID   string
+		machine     string
+		expected    Info
+		expectedErr string
+	}{
+		{
+			desc:      "Valid machine target",
+			modelUUID: "8419cd78-4993-4c3a-928e-c646226beeee",
+			machine:   "1",
+			expected: Info{
+				target:    MachineTarget,
+				modelUUID: "8419cd78-4993-4c3a-928e-c646226beeee",
+				machine:   1,
+			},
+		},
+		{
+			desc:        "Invalid model UUID",
+			modelUUID:   "invalid-uuid",
+			machine:     "1",
+			expectedErr: ".*invalid model UUID.*",
+		},
+		{
+			desc:        "Invalid machine number",
+			modelUUID:   "8419cd78-4993-4c3a-928e-c646226beeee",
+			machine:     "-1",
+			expectedErr: "invalid machine number: -1",
+		},
+		{
+			desc:        "Invalid machine number: nested container",
+			modelUUID:   "8419cd78-4993-4c3a-928e-c646226beeee",
+			machine:     "0/lxd/0",
+			expectedErr: "container machine not supported",
+		},
+	}
+
+	for i, tC := range testCases {
+		c.Logf("test %d: %s", i, tC.desc)
+		res, err := NewInfoMachineTarget(tC.modelUUID, tC.machine)
+		if tC.expectedErr == "" {
+			c.Assert(err, gc.IsNil)
+			c.Assert(res, gc.DeepEquals, tC.expected)
 		} else {
 			c.Assert(err, gc.ErrorMatches, tC.expectedErr)
 		}
```

Comment:
This will allow all test cases to run and only fail at the end if one of the checks fail. `c.Check` returns a bool, which can be helpful in this code path.

```suggestion
		if tC.expectedErr == "" {
			if c.Check(err, gc.IsNil) {
			      c.Check(res, gc.DeepEquals, tC.expected)
			}
		} else {
			c.Check(err, gc.ErrorMatches, tC.expectedErr)
		}
```